### PR TITLE
Add Item Status Edit Actions

### PR DESCRIPTION
**Pull request by YanaDePauw**
This PR includes the majority of the functionality on the Edit Item Status page:

- Withdraw and Reinstate item
- Make Item Private and Public
- Permanently Delete item

The Withdraw/Reinstate and Private/Public actions will send a PATCH request to the REST API and wait for a response before redirecting the user back to the Edit Item Status page.

The Permanently delete action will send a DELETE request to the server and redirect the user to the homepage on success as the item itself will have been deleted.

To test the functionality, navigate to an item page, add "/edit" at the url, and click the Withdraw/Reinstate, Private/Public or Permanently delete button. 

NOTE:
The tombstone page to be displayed for a withdrawn item is not yet present. There is still a dependency missing in the REST API to fully support this. Issues have been created for both the UI and the REST API: 
https://github.com/DSpace/dspace-angular/issues/346 
https://jira.duraspace.org/browse/DS-4125

### CODE REVIEW

File: src/app/+item-page/edit-item-page/edit-item-page.module.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import {ItemOperationComponent} from './item-operation/item-operation.component';
+import {ModifyItemOverviewComponent} from './modify-item-overview/modify-item-overview.component';
+import {ItemWithdrawComponent} from './item-withdraw/item-withdraw.component';
+import {ItemReinstateComponent} from './item-reinstate/item-reinstate.component';
+import {AbstractSimpleItemActionComponent} from './simple-item-action/abstract-simple-item-action.component';
+import {ItemPrivateComponent} from './item-private/item-private.component';
+import {ItemPublicComponent} from './item-public/item-public.component';
+import {ItemDeleteComponent} from './item-delete/item-delete.component';
+
+@NgModule({
```
Comment (tdonohue) [inline src/app/+item-page/edit-item-page/edit-item-page.module.ts]: Please add TypeDocs to this new Module

File: src/app/+item-page/edit-item-page/edit-item-page.routing.module.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import {ItemPublicComponent} from './item-public/item-public.component';
+import {ItemDeleteComponent} from './item-delete/item-delete.component';
+
+const ITEM_EDIT_WITHDRAW_PATH = 'withdraw';
+const ITEM_EDIT_REINSTATE_PATH = 'reinstate';
+const ITEM_EDIT_PRIVATE_PATH = 'private';
+const ITEM_EDIT_PUBLIC_PATH = 'public';
+const ITEM_EDIT_DELETE_PATH = 'delete';
+
+@NgModule({
```
Comment (tdonohue) [inline src/app/+item-page/edit-item-page/edit-item-page.routing.module.ts]: Please add TypeDocs to this new module

File: src/app/+item-page/edit-item-page/item-operation/itemOperation.model.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+export class ItemOperation {
```
Comment (tdonohue) [inline src/app/+item-page/edit-item-page/item-operation/itemOperation.model.ts]: Please add TypeDocs to this new class

File: src/app/+item-page/edit-item-page/simple-item-action/abstract-simple-item-action.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+import {By} from '@angular/platform-browser';
+import {RestResponse} from '../../../core/cache/response-cache.models';
+import {of as observableOf} from 'rxjs';
+import {getItemEditPath} from '../../item-page-routing.module';
+
+@Component({
+  selector: 'ds-simple-action',
+  templateUrl: './abstract-simple-item-action.component.html'
+})
+export class MySimpleItemActionComponent extends AbstractSimpleItemActionComponent {
```
Comment (tdonohue) [inline src/app/+item-page/edit-item-page/simple-item-action/abstract-simple-item-action.component.spec.ts]: Please add TypeDocs to this new Component

**Comment by YanaDePauw** [comment]
Added additional typedocs. 

Concerning the authentication issue, I only have this issue locally when using dev settings when preboot is set to false `preboot: false`, because the login state does not seem to be saved in that mode. When running in production mode, or with preboot set to true, I do not have the issue.

**Comment by artlowel** [comment]
@tdonohue following those instructions I can replicate it as well. A possible explanation could be that REST PR https://github.com/DSpace/DSpace/pull/2207 hasn't yet been applied to https://dspace7.4science.it/dspace-spring-rest/ yet. @abollini would you mind verifying that?

**Comment by tdonohue** [comment]
@YanaDePauw and @artlowel : Apologies for the (very long) delay in getting back to testing this again.  After getting the demo REST API updated, this all works as expected. I'm no longer seeing the re-prompting for login.

So, this looks great 👍  and I'll merge it.

**Pull request merged**